README.md

```diff
@@ -11,53 +11,238 @@ Pour faire ce codelab, vous avez besoin des outils suivants :
+    name: 'Blastoise',
+    image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/10.png',
+}, {
+    name: 'Caterpie',
+    image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
+}]
+```
+```html
+{{ pokemons[0].name }}
+```
+N'oubliez pas de mettre à jour App.vue pour importer ce nouveau composant.
+
+- On va maintenant extraire les cards de chaque pokemon dans un composant dédié `card.vue`. Ce composant prendre deux propriétées:
+    - `name` qui sera le nom du pokemon
+    - `image`qui sera l'URL de l'image à affichée
+Ré-écriver le composant `result.vue` pour utiliser ce nouveau composant cards
+
+- *(bonus)* inscrivez le composant `result.vue` de manière global.
+
+
+### PW3 - Directives
+
+
+Le composant `search.vue` est actuellement qu'un simple champ de formulaire. Nous allons le rendre un peu plus intéractif en lui ajoutant certains comportement:
+- Utilisez la directive `v-model` pour binder le champ `input` à une variable du composant
+- Ajouter un handler sur l'événement `keyup` qui va permettre au composant d'émettre un événement `onSearchChange` utilisable par un composant parent
+- *(bonus)* Ajoutez lodash pour que l'événément onSearchChange ne soit émit qu'après un certain délai de saisi (`debounce`)
+
+Maitenant, nous allons mettre à jour le composant `result.vue` pour qu'il affiche une liste filtrée:
+- Affiche la liste des pokemons en utilisant la directive `v-for`
+- Ajoutez une propriété `criteria` qui sera la chaîne de caractère qui permettra de filtrer nos pokemon
+- Ajouter une condition de rendu du composant en utilisant la directive `v-if`. Le méthode de filtre sera la suivante
+```javascript
+const lowerSearch = this.criteria.toLowerCase();
+const lowerPokemon = pokemon.name.toLowerCase();
+return lowerPokemon.includes(lowerSearch);
+```
+
+Nous allons maitenant lier les deux composants via `App.vue`.
+- Ajoutez un handler de l'événement `onSearchChange` qui va récupérer la valeur de recherche et la stocker.
+- Passer cette donnée dans le composant result via la nouvelle propriété `criteria`.
+
+### PW4 - Filtres
+
+Mettez à jour la liste des pokemons avec le code suivant :
+```javascript
+pokemons: [{
+    id: 1,
+    name: 'bulbasaur',
+}, {
+    id: 2,
+    name: 'ivysaur',
+}, {
+    id: 3,
+    name: 'venusaur',
+}, {
+    id: 4,
+    name: 'charmander',
+}, {
+    id: 5,
+    name: 'charmeleon',
+}, {
+    id: 6,
+    name: 'charizard',
+}, {
+    id: 8,
+    name: 'squirtle',
+}, {
+    id: 9,
+    name: 'wartortle',
+}, {
+    id: 10,
+    name: 'blastoise',
+}, {
+    id: 11,
+    name: 'caterpie',
+}]
+```
+- Créer un filtre custom pour pemettre d'afficher la première lettre du nom du Pokemon en majuscule
+- Créer une computed property pour calculer l'URL de l'image du pokemon à partir de son ID.
+
+### PW5 - Ressources
+
+Nous allons maintenant récupérer la liste des pokemons depuis une API: https://pokeapi.co
+
+- Installez la librairie axios

```

Comment:
~installez~`installer` ? Infinitive has been used just below, so for harmonization